### DISCUSSION TITLE
How can I run acquisition function seperately to find the next input x??

### DISCUSSION THREAD

**Discussion by chibai** [optuna/optuna#4007] [Q&A]
I wish I can explain my main idea like this.
```python
def objective(trial):
    j = trial.suggest_float('j', -1, 1)
    return j**2 + j + 1

sudty = optuna.create_study()
study.optimize(objective, n_trials = 0)
next_input = study.acquisition_function()
```
I want to control the whole procedures of bayesian optimization

**Comment by nzw0301** [comment]
Unfortunately, Optuna doesn't provide such an interface to access the acquisition functions. Thus this way

> I want to control the whole procedures of bayesian optimization

sounds difficult for Optuna.

**Comment by Kertin** [comment]
same question

**Reply by nzw0301** [reply]
Answered in https://github.com/optuna/optuna/discussions/4007#discussioncomment-3666481
